File: Back/Models/Dogadjaj.cs
using KrvNijeVoda.Models;

namespace KrvNijeVoda.Back.Models
{ 
    public enum TipDogadjaja {
        Bitka,
        Rat,
        Ustanak, 
        Sporazum,
        Savez,
        Dokument
    }
    public class Dogadjaj {
        public Guid ID { get; set; }
        public string Ime { get; set; }//unique
        public TipDogadjaja Tip { get; set; }
        public Godina? Godina { get; set; }//mozda nekad null sad ne
        public string? Lokacija { get; set; }//STAVI VEZU SA ZEMLJOM
        //public Lokacija? Lokacija { get; set; }
        //public List<Zemlja> Ucesnici { get; set; } = new List<Zemlja>();
        public string? Tekst { get; set; }
        //ne mora da ga nasledi rat nego samo da ima nullable GOdina do i taman bitka ne mora to da ima
        //bolje da nasledi da ne bude mng null propertya i zbog prosirljivosti
    }
    
}

File: front/src/types/dogadjaj.ts
import type { Godina } from "./godina";

export type TipDogadjaja =
    | "Bitka"
    | "Rat"
    | "Ustanak"
    | "Sporazum"
    | "Savez"
    | "Dokument"
    | "Opsada";

export interface DogadjajBase {
    id: string;
    tip: TipDogadjaja;
    ime: string;
    godina?: Godina | null;
    lokacija?: string | null;
    tekst: string;
}

export interface Bitka extends DogadjajBase {
    tip: "Bitka";
    pobednik: string;
    rat?: string;
    brojZrtava: number;
}

export interface Rat extends DogadjajBase {
    tip: "Rat";
    godinaDo?: Godina | null;
    bitke: string[];
    pobednik: string;
}

export interface OstaliDogadjaj extends DogadjajBase {
    tip: Exclude<TipDogadjaja, "Bitka" | "Rat">;
}

export type DogadjajUnion = Bitka | Rat | OstaliDogadjaj;
